core/src/preferences/PreferencesBackend.cpp

```diff
@@ -194,17 +194,47 @@ namespace ledger {
             unsetEncryption();
 
             auto emptySalt = std::string("");
-            auto pref = getPreferences("__core");
-            auto salt = pref->getString(ENCRYPTION_SALT_KEY, emptySalt);
+            auto saltKey = std::vector<uint8_t>(ENCRYPTION_SALT_KEY.cbegin(), ENCRYPTION_SALT_KEY.cend());
+            auto salt = get(saltKey).value_or(emptySalt);
 
             if (salt == emptySalt) {
-                // we don’t have a proper salt; create one and persist it for future use
+                leveldb::WriteBatch batch;
+
+                // we don’t have a proper salt; create one
                 salt = createNewSalt(rng);
-                pref->editor()->putString(ENCRYPTION_SALT_KEY, salt)->commit();
-            }
 
-            // create the AES cipher
-            _cipher = AESCipher(rng, password, salt, PBKDF2_ITERS);
+                // create the AES cipher
+                _cipher = AESCipher(rng, password, salt, PBKDF2_ITERS);
+
+                // add the salt to the batch to be written
+                auto saltData = std::vector<uint8_t>(salt.cbegin(), salt.cend());
+                auto noCipher = Option<AESCipher>::NONE;
+                putPreferencesChange(batch, noCipher, PreferencesChange(PreferencesChangeType::PUT_TYPE, saltKey, saltData));
+
+                // iterate through all records already present and update them
+                auto it = std::unique_ptr<leveldb::Iterator>(_db->NewIterator(leveldb::ReadOptions()));
+                for (it->SeekToFirst(); it->Valid(); it->Next()) {
+                    // read the clear value
+                    auto value = it->value().ToString();
+                    auto plain = std::vector<uint8_t>(std::begin(value), std::end(value));
```

Comment:
sometimes you use std::begin(x), sometimes x.cbegin(), lets steek to one style + we shouldn't explicitly specify a namespace `std::` in this case, the compiler should resolve the name based on the argument (so fo collections from stl it would be std and for our own collections it woulb be our own begin, that can be in ledger::core name)

Reply:
I’ll stick to the postfix version for now, nice catch.